### ISSUE TITLE
Excessive Logcat debug output: "Received NDK message"

### ISSUE THREAD

**Issue by faithfracture** [bugsnag/bugsnag-android#919]
Bugsnag is dumping out an excessive amount of debug logging for NDK breadcrumb messages. My logs are getting filled up with debug messages that look like:

`D/Bugsnag: Received NDK message com.bugsnag.android.StateEvent$AddBreadcrumb@3fefb51`

This is not useful information, and I have my own native debug logging that shows me _much_ more detailed information, which I can enable & disable as needed. I _really_ don't need the extraneous messages from Bugsnag.

(If there's a way to squelch this I have not been able to figure it out)

### Expected behavior
I expect to be able to configure the log level for Bugsnag NDK breadcrumb message receipts, rather than have it output Debug logs 100% of the time.

### Observed behavior
Bugsnag always outputs debug logs for NDK messages

### Steps to reproduce
Add some breadcrumbs in native library code. 

### Version
Bugsnag - 5.0.0
Bugsnag Gradle Plugin - 4.7.5

**Comment by snmaynard** [comment]
It looks like that is a debug level message from the library, so it shouldnt be output by default. You can configure the logger to do whatever you like however -> https://docs.bugsnag.com/platforms/android/configuration-options/#logger

**Comment by faithfracture** [comment]
> it shouldnt be output by default

Does that mean it's a something in Bugsnag that needs to be addressed by the Bugsnag team, or that I've got something goofed on my end?

It sounds like my option is to use the default logging behavior or provide my own implementation? I was kind of hoping for something more along the lines of:

```kotlin
val config = Configuration.load(this)
config.logLevel = Logger.Debug
Bugsnag.start(this, config)
```

So we can just turn debug logging on & off as needed.

**Comment by snmaynard** [comment]
I think that in the short term you are best off implementing your own logger, but this looks like something we should do a better job of handling in the library - so it depends on how urgent it is for you.

**Comment by faithfracture** [comment]
Ok thanks - just wanted to clarify.

I'm not in a rush - I have no problem adding my own logger.

What's interesting to me, looking at the log, is that the point at which this log is being generated is not in any native / NDK code (due to an implementation detail on my side I haven't been able to actually integrate the NDK Bugsnag logging API with my existing debug logger). I have my own wrapper around the standard Android logger that looks like this

```kotlin
class Logger @JvmOverloads constructor(private val tag: String, private val level: Int = INFO) {
    companion object {
        // Public constants so we don't have to include 2 different classes for logging
        const val VERBOSE = Log.VERBOSE
        const val DEBUG = Log.DEBUG
        const val INFO = Log.INFO
        const val WARN = Log.WARN
        const val ERROR = Log.ERROR
    }

//... other logger impl

    fun i(msg: String) {
        if (level <= INFO) {
            val metadata = hashMapOf<String, Any>(
                    "message" to msg,
                    "level" to "Info"
            )
            Bugsnag.leaveBreadcrumb(tag, metadata, BreadcrumbType.LOG)
            Log.i(tag, msg)
        }
    }

//... etc.
}
```

One place where I'm specifically seeing this output is from a Kotlin class that's just calling (e.g.)

```kotlin
class MyClass {
    companion object {
        val log = Logger(MyClass::class.java.simpleName)
    }

    fun someFunc() {
        log.i("Something happened")
    }
}

```

In Logcat I'll see:

>2020-08-06 14:22:31.805 17131-17131/com.example I/MyClass: Something happened
>2020-08-06 14:22:31.805 17131-17131/com.example D/Bugsnag: Received NDK message com.bugsnag.android.StateEvent$AddBreadcrumb@4e1f78f

I don't know if that's helpful in anyway, but I figured I'd share it.

**Comment by faithfracture** [comment]
FWIW, I set a breakpoint in my custom logger implementation and caught the place this is happening:

`NativeBridge.isInvalidMessage`
```kotlin
    private fun isInvalidMessage(msg: Any?): Boolean {
        if (msg !is StateEvent) {
            return true
        }
        if (!installed.get() && msg !is Install) {
            logger.w("Received message before INSTALL: $msg")
            return true
        }

        logger.d(String.format("Received NDK message %s", msg)) // <--- Here
        return false
    }
```

**Comment by mattdyoung** [comment]
The excessive logging has been removed as part of https://github.com/bugsnag/bugsnag-android/pull/1238 released in [v5.9.2](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.9.2)
